File: core/src/nl/t64/game/rpg/components/party/ResourceType.java
package nl.t64.game.rpg.components.party;

import lombok.AllArgsConstructor;


@AllArgsConstructor
enum ResourceType implements SuperEnum {

    GOLD("Gold"),
    HERB("Herb"),
    SPICE("Spice"),
    GEMSTONE("Gemstone");

    final String title;

    @Override
    public String getTitle() {
        return title;
    }

}

File: core/src/nl/t64/game/rpg/components/party/spells/ResourceType.kt
package nl.t64.game.rpg.components.party.spells

import nl.t64.game.rpg.components.party.SuperEnum


enum class ResourceType(override val title: String) : SuperEnum {

    GOLD("Gold"),
    HERB("Herb"),
    SPICE("Spice"),
    GEMSTONE("Gemstone");

}
